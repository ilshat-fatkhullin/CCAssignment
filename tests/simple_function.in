/* Simple function with 2 parameters that perform addition and multiplication operations. */

static int sum(int a, int b) {
    int c; int result;
    result = a + b;
    c = a * b;
    return result;
    }