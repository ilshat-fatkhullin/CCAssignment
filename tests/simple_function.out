{
    "compilationUnit": [
        {
            "externalDeclaration": {
                "functionDefinition": [
                    {
                        "declarationSpecifiers": [
                            {
                                "declarationSpecifier": {
                                    "storageClassSpecifier": "static"
                                }
                            },
                            {
                                "declarationSpecifier": {
                                    "typeSpecifier": "int"
                                }
                            }
                        ]
                    },
                    {
                        "declarator": {
                            "directDeclarator": [
                                {
                                    "directDeclarator": "sum"
                                },
                                "(",
                                {
                                    "parameterTypeList": {
                                        "parameterList": [
                                            {
                                                "parameterList": {
                                                    "parameterDeclaration": [
                                                        {
                                                            "declarationSpecifiers": {
                                                                "declarationSpecifier": {
                                                                    "typeSpecifier": "int"
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "declarator": {
                                                                "directDeclarator": "a"
                                                            }
                                                        }
                                                    ]
                                                }
                                            },
                                            ",",
                                            {
                                                "parameterDeclaration": [
                                                    {
                                                        "declarationSpecifiers": {
                                                            "declarationSpecifier": {
                                                                "typeSpecifier": "int"
                                                            }
                                                        }
                                                    },
                                                    {
                                                        "declarator": {
                                                            "directDeclarator": "b"
                                                        }
                                                    }
                                                ]
                                            }
                                        ]
                                    }
                                },
                                ")"
                            ]
                        }
                    },
                    {
                        "compoundStatement": [
                            "{",
                            {
                                "blockItemList": [
                                    {
                                        "blockItem": {
                                            "declaration": [
                                                {
                                                    "declarationSpecifiers": {
                                                        "declarationSpecifier": {
                                                            "typeSpecifier": "int"
                                                        }
                                                    }
                                                },
                                                {
                                                    "initDeclaratorList": {
                                                        "initDeclarator": {
                                                            "declarator": {
                                                                "directDeclarator": "c"
                                                            }
                                                        }
                                                    }
                                                },
                                                ";"
                                            ]
                                        }
                                    },
                                    {
                                        "blockItem": {
                                            "declaration": [
                                                {
                                                    "declarationSpecifiers": {
                                                        "declarationSpecifier": {
                                                            "typeSpecifier": "int"
                                                        }
                                                    }
                                                },
                                                {
                                                    "initDeclaratorList": {
                                                        "initDeclarator": {
                                                            "declarator": {
                                                                "directDeclarator": "result"
                                                            }
                                                        }
                                                    }
                                                },
                                                ";"
                                            ]
                                        }
                                    },
                                    {
                                        "blockItem": {
                                            "statement": {
                                                "expressionStatement": [
                                                    {
                                                        "expression": {
                                                            "assignmentExpression": [
                                                                {
                                                                    "unaryExpression": {
                                                                        "postfixExpression": {
                                                                            "primaryExpression": "result"
                                                                        }
                                                                    }
                                                                },
                                                                {
                                                                    "assignmentOperator": "="
                                                                },
                                                                {
                                                                    "assignmentExpression": {
                                                                        "conditionalExpression": {
                                                                            "logicalOrExpression": {
                                                                                "logicalAndExpression": {
                                                                                    "inclusiveOrExpression": {
                                                                                        "exclusiveOrExpression": {
                                                                                            "andExpression": {
                                                                                                "equalityExpression": {
                                                                                                    "relationalExpression": {
                                                                                                        "shiftExpression": {
                                                                                                            "additiveExpression": [
                                                                                                                {
                                                                                                                    "additiveExpression": {
                                                                                                                        "multiplicativeExpression": {
                                                                                                                            "castExpression": {
                                                                                                                                "unaryExpression": {
                                                                                                                                    "postfixExpression": {
                                                                                                                                        "primaryExpression": "a"
                                                                                                                                    }
                                                                                                                                }
                                                                                                                            }
                                                                                                                        }
                                                                                                                    }
                                                                                                                },
                                                                                                                "+",
                                                                                                                {
                                                                                                                    "multiplicativeExpression": {
                                                                                                                        "castExpression": {
                                                                                                                            "unaryExpression": {
                                                                                                                                "postfixExpression": {
                                                                                                                                    "primaryExpression": "b"
                                                                                                                                }
                                                                                                                            }
                                                                                                                        }
                                                                                                                    }
                                                                                                                }
                                                                                                            ]
                                                                                                        }
                                                                                                    }
                                                                                                }
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            ]
                                                        }
                                                    },
                                                    ";"
                                                ]
                                            }
                                        }
                                    },
                                    {
                                        "blockItem": {
                                            "statement": {
                                                "expressionStatement": [
                                                    {
                                                        "expression": {
                                                            "assignmentExpression": [
                                                                {
                                                                    "unaryExpression": {
                                                                        "postfixExpression": {
                                                                            "primaryExpression": "c"
                                                                        }
                                                                    }
                                                                },
                                                                {
                                                                    "assignmentOperator": "="
                                                                },
                                                                {
                                                                    "assignmentExpression": {
                                                                        "conditionalExpression": {
                                                                            "logicalOrExpression": {
                                                                                "logicalAndExpression": {
                                                                                    "inclusiveOrExpression": {
                                                                                        "exclusiveOrExpression": {
                                                                                            "andExpression": {
                                                                                                "equalityExpression": {
                                                                                                    "relationalExpression": {
                                                                                                        "shiftExpression": {
                                                                                                            "additiveExpression": {
                                                                                                                "multiplicativeExpression": [
                                                                                                                    {
                                                                                                                        "multiplicativeExpression": {
                                                                                                                            "castExpression": {
                                                                                                                                "unaryExpression": {
                                                                                                                                    "postfixExpression": {
                                                                                                                                        "primaryExpression": "a"
                                                                                                                                    }
                                                                                                                                }
                                                                                                                            }
                                                                                                                        }
                                                                                                                    },
                                                                                                                    "*",
                                                                                                                    {
                                                                                                                        "castExpression": {
                                                                                                                            "unaryExpression": {
                                                                                                                                "postfixExpression": {
                                                                                                                                    "primaryExpression": "b"
                                                                                                                                }
                                                                                                                            }
                                                                                                                        }
                                                                                                                    }
                                                                                                                ]
                                                                                                            }
                                                                                                        }
                                                                                                    }
                                                                                                }
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            ]
                                                        }
                                                    },
                                                    ";"
                                                ]
                                            }
                                        }
                                    },
                                    {
                                        "blockItem": {
                                            "statement": {
                                                "jumpStatement": [
                                                    "return",
                                                    {
                                                        "expression": {
                                                            "assignmentExpression": {
                                                                "conditionalExpression": {
                                                                    "logicalOrExpression": {
                                                                        "logicalAndExpression": {
                                                                            "inclusiveOrExpression": {
                                                                                "exclusiveOrExpression": {
                                                                                    "andExpression": {
                                                                                        "equalityExpression": {
                                                                                            "relationalExpression": {
                                                                                                "shiftExpression": {
                                                                                                    "additiveExpression": {
                                                                                                        "multiplicativeExpression": {
                                                                                                            "castExpression": {
                                                                                                                "unaryExpression": {
                                                                                                                    "postfixExpression": {
                                                                                                                        "primaryExpression": "result"
                                                                                                                    }
                                                                                                                }
                                                                                                            }
                                                                                                        }
                                                                                                    }
                                                                                                }
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }
                                                    },
                                                    ";"
                                                ]
                                            }
                                        }
                                    }
                                ]
                            },
                            "}"
                        ]
                    }
                ]
            }
        },
        "<EOF>"
    ]
}
